zipline/gens/tradesimulation.py

```diff
@@ -100,6 +101,16 @@ def transform(self, stream_in):
                 self.simulation_dt = date
                 self.on_dt_changed(date)
 
+                # removing expired futures
+                for sid in self.current_data.keys():
+                    try:
+                        if self.env.asset_finder.retrieve_asset(sid).end_date \
+                                < self.simulation_dt:
+                            self.current_data.__delitem__(sid)
+                    except (AttributeError, TypeError, ValueError,
```

Comment:
AttributeError - when the asset is a Future and has no end_date
SidNotFound - fails in a test where the sid for an asset is changed
TypeError - when end_date=None
ValueError - not sure. gotta recheck that one.